packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -440,11 +441,105 @@ class RenderParagraph extends RenderBox {
   @override
   void describeSemanticsConfiguration(SemanticsConfiguration config) {
     super.describeSemanticsConfiguration(config);
-    config
-      ..label = text.toPlainText()
-      ..textDirection = textDirection;
+    bool hasRecognizers = false;
+    text.visitTextSpan((TextSpan span) {
+      if (span.recognizer != null) {
+        hasRecognizers = true;
+        return false;
+      }
+      return true;
+    });
+    if (hasRecognizers) {
+      config.explicitChildNodes = true;
+      config.isSemanticBoundary = true;
+    } else {
+      config.label = text.toPlainText();
+      config.textDirection = textDirection;
+    }
   }
 
+  @override
+  void assembleSemanticsNode(SemanticsNode node, SemanticsConfiguration config, Iterable<SemanticsNode> children) {
+    final List<SemanticsNode> newChildren = <SemanticsNode>[];
+    StringBuffer buffer;
+    double order = 0.0;
+    int start = 0;
+    int end = 0;
+    // traverse all spans and collect them into adjacent compatible nodes.
+    text.visitTextSpan((TextSpan span) {
+      if (span.recognizer != null) {
+        if (buffer != null) {
+          final SemanticsNode child = new SemanticsNode();
+          final SemanticsConfiguration childConfig = new SemanticsConfiguration();
+          child.rect = _getRectForSpan(start, end);
+          childConfig
+            ..label = buffer.toString()
+            ..textDirection = textDirection
+            ..sortKey = new OrdinalSortKey(order++);
```

Comment:
this might be clearer as just passing order here, and then having a separate statement that says `order += 1`

Reply:
Done